go/binlog/gomysql_reader.go

```diff
@@ -120,6 +126,11 @@ func (this *GoMySQLReader) handleRowsEvent(ev *replication.BinlogEvent, rowsEven
 		// In reality, reads will be synchronous
 		entriesChannel <- binlogEntry
 	}
+	if this.migrationContext.UseGTIDs {
+		this.currentCoordinatesMutex.Lock()
+		defer this.currentCoordinatesMutex.Unlock()
```

Comment:
One thing that might be better is to simply move this line to after 132 without using `defer`. This is because the defer will wait until the method ends to actually release the mutex lock, which is often longer than needed.  Since there's literally only one assignment operation between the lock/unlock then there's really no chance for the method to panic, and there's no branches which include a return after the lock.